orbit_predictor/predictors/pass_iterators.py

```diff
@@ -201,84 +201,68 @@ def _find_los(self, tca):
 class SmartLocationPredictor(BaseLocationPredictor):
 
     def iter_passes(self):
-        current_date = self.start_date
-        while True:
-            valid, aos, tca, los, max_elevation = self._find_next_pass(current_date)
-            current_date = los
+        # Explore all values of t every 3 minutes
+        t_values = list(range(0, int((self.limit_date - self.start_date).total_seconds()), 180))
+        elev_values = []
+        for delta_seconds in t_values:
+            elev_values.append(self._elevation(delta_seconds))
 
-            if self.limit_date is not None and current_date > self.limit_date:
-                break
-            else:
-                if valid:
-                    yield PredictedPass(
-                        self.location,
-                        self.predictor.sate_id,
-                        max_elevation_deg=degrees(max_elevation),
-                        aos=aos,
-                        los=los,
-                        duration_s=(los - aos).total_seconds(),
-                        max_elevation_position=self.predictor.get_position(tca),
-                        max_elevation_date=tca,
-                     )
+        peaks_idx, _ = find_peaks(elev_values)
 
-    def _elevation_at(self, when_utc):
+        for peak_idx in peaks_idx:
+            if elev_values[peak_idx] <= 0:
+                continue
+            else:
+                t_approximate_tca = t_values[peak_idx]
+                period_s = orbital_period(self.predictor.mean_motion) * 60
+
+                aos, tca, los, max_elevation = self._refine_pass(t_approximate_tca, period_s)
+
+                yield PredictedPass(
+                    self.location,
+                    self.predictor.sate_id,
+                    max_elevation_deg=degrees(max_elevation),
+                    aos=aos,
+                    los=los,
+                    duration_s=(los - aos).total_seconds(),
+                    max_elevation_position=self.predictor.get_position(tca),
+                    max_elevation_date=tca,
+                )
+
+    def _elevation(self, delta_seconds):
+        when_utc = self.start_date + dt.timedelta(seconds=delta_seconds)
         position = self.predictor.get_only_position(when_utc)
         return self.location.elevation_for(position)
 
-    def _find_next_pass(self, start_date):
-        def elevation(delta_seconds):
-            return self._elevation_at(start_date + dt.timedelta(seconds=delta_seconds))
-
-        start_elevation = self._elevation_at(start_date)
-        period_s = orbital_period(self.predictor.mean_motion) * 60
-
-        # Find date for maximum elevation within the next orbital period
-        # NOTE: This is the most expensive operation
+    def _refine_pass(self, t_approximate_tca, period_s):
+        # AOS must be between half the previous period and the approximate TCA
+        t_aos = root_scalar(
```

Comment:
Just curious, why are we not using `brentq`? is this other method faster or better?

Reply:
We are indeed using `brentq` :) But I'll make it explicit.

https://github.com/scipy/scipy/blob/01d8bfb6f239df4ce70c799b9b485b53733c9911/scipy/optimize/_root_scalar.py#L219-L220

Reply:
ohh, I see. Thanks!